plugins/src/hashlookup/select.go

```diff
@@ -0,0 +1,241 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/blastrain/vitess-sqlparser/sqlparser"
+)
+
+// If the where is empty, need to check whether to agg or not
+func checkNeedAgg(sqlSelect sqlparser.SelectExprs) bool {
+	for _, v := range sqlSelect {
+		expr, ok := v.(*sqlparser.AliasedExpr)
+		if !ok {
+			// No need to handle, star expression * just skip is ok
+			continue
+		}
+
+		// TODO: more precise
+		if _, ok := expr.Expr.(*sqlparser.FuncExpr); ok {
+			return true
+		}
+	}
+
+	return false
+}
+
+func handleSelectWhereComparisonExpr(expr *sqlparser.Expr, topLevel bool, parent *sqlparser.Expr) ([][2]string, error) {
+	comparisonExpr := (*expr).(*sqlparser.ComparisonExpr)
+	colName, ok := comparisonExpr.Left.(*sqlparser.ColName)
+
+	if !ok {
+		return nil, errors.New("Invalid comparison expression, the left must be a column name")
+	}
+
+	colNameStr := sqlparser.String(colName)
+	colNameStr = strings.Replace(colNameStr, "`", "", -1)
+	rightIntf, err := buildComparisonExprRightStr(comparisonExpr.Right)
+	if err != nil {
+		return nil, err
+	}
+
+	fields := [][2]string{}
+
+	switch comparisonExpr.Operator {
+	case "=":
+		fields = append(fields, [2]string{colNameStr, fmt.Sprintf("%s", rightIntf)})
+	default:
+		return nil, errors.New("'=' operator is supported only")
+	}
+
+	return fields, nil
+}
+
+func handleSelectWhereAndExpr(expr *sqlparser.Expr, topLevel bool, parent *sqlparser.Expr) ([][2]string, error) {
+	andExpr := (*expr).(*sqlparser.AndExpr)
+	leftExpr := andExpr.Left
+	rightExpr := andExpr.Right
+
+	leftStr, err := handleSelectWhere(&leftExpr, false, expr)
+	if err != nil {
+		return nil, err
+	}
+
+	rightStr, err := handleSelectWhere(&rightExpr, false, expr)
+	if err != nil {
+		return nil, err
+	}
+
+	fields := append(leftStr, rightStr...)
+	if topLevel {
+		fields = append(fields, [2]string{"operator", "and"})
+	}
+
+	return fields, nil
+}
+
```

Comment:
/*
 * Handle "expression OR expression".
 *
 * Receives:
 * expr     - SQL expression to process
 * topLevel - whether it's a top level expression
 * parent   - container of the expression
 */